ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorThread.java

```diff
@@ -61,6 +61,14 @@ public abstract class CompactorThread extends Thread implements Configurable {
   protected String hostName;
   protected String runtimeVersion;
 
+  //Time threshold for compactor thread log
+  //In milliseconds:
+  protected Integer MAX_WARN_LOG_TIME = 1200000; //20 min
+
+  protected long checkInterval = 0;
+
+  protected enum CompactorThreadType {INITIATOR, WORKER, CLEANER}
```

Comment:
minor: can't we use package-private visibility?

Reply:
Initial thought for this was to be used by just the subclasses, but on a second thought makes sense for it to be package-private for other classes in package to use this as well in future. Will change